### ISSUE TITLE
MenuItems in ContextMenu on Tray Icon Using Commands Always Disabled

### ISSUE THREAD

**Issue by uyaem** [hardcodet/wpf-notifyicon#23]
**CS**
```
public static RoutedCommand ChooseFolderCommand = new RoutedCommand();

private void ChooseFolderCommandExecuted(object sender, ExecutedRoutedEventArgs e)
{
	// ...
}
```

**XAML**
```
<Window.CommandBindings>
    <CommandBinding Command="local:MainWindow.ChooseFolderCommand" Executed="ChooseFolderCommandExecuted" />
</Window.CommandBindings>

<tb:TaskbarIcon x:Name="trayMenu" IconSource="/Assets/quill-64.ico">
    <tb:TaskbarIcon.ContextMenu>
        <ContextMenu Opened="ContextMenu_Opened">
            <MenuItem Header="Choose folder" Command="local:MainWindow.ChooseFolderCommand" />
        </ContextMenu>
    </tb:TaskbarIcon.ContextMenu>
</tb:TaskbarIcon>
```

**Noteworthy**
* The Command gets never disabled (no CanExecute), yet when debugging, the `IsEnabled` property of the MenuItem remains false.
* The same Command is used in a `Menu` and works as expected.
* In order to resolve the issue, I had temporarily attached a `CanExecute`-method to my CommandBinding, which doesn't get called ever for the tray icon's context menu.  This remains the case even when calling `CommandManager.InvalidateRequerySuggested()` in e.g. the `Opened` event of the ContextMenu.
* The above works fine when using a Click event hander instead
* Others seem to have the same issue, see [this question](https://stackoverflow.com/questions/30963641/wpf-notifyicon-context-menu-using-applicationcommands-is-disabled) on StackOverflow

**Comment by maxpavlov** [comment]
Since this is a major drawback affecting many who follow the standard command approach for context menus, could you please elaborate on how you managed to resolve this via CanExecute by providing the example? Would be really appreciated.

**Comment by maxpavlov** [comment]
Oh, I see, you @uyaem probably meant that in order to troubleshoot the issue you've attached the `CanExecute` method, but it didn't yield any results nonetheless, as it never gets call from what I see.

**Comment by maxpavlov** [comment]
Just for anyone who comes across this - I was only able to get this to work using the trivial (non routed commands) from the example application in the repository. I had to inject the IoC into the command handler via a parameter allowing the commands to be kind of self-sufficient, otherwise, as per example, they can only react within the Visual Tree which is pretty useless. So if you need commands to work in the Tray Icon context menu - use the approach from here: https://github.com/hardcodet/wpf-notifyicon/tree/master/src/NotifyIconWpf.Sample.ShowCases/Tutorials/04%20-%20ContextMenus

**Comment by Lakritzator** [comment]
I'm horrible at WPF, commands and "bubbling" events... I have seen many weird things, and usually just take the quirks and their solutions for granted...

I believe the context menu is completely separate from the normal visual tree (it's a popup with it's own root), and thus also has issues with events, commands and even styles.

I'm not sure if this can be made better, but the [mentioned tutorial](https://github.com/hardcodet/wpf-notifyicon/tree/master/src/NotifyIconWpf.Sample.ShowCases/Tutorials/04%20-%20ContextMenus) seems to help.

I also found this here: https://stackoverflow.com/questions/59698021/wpf-context-menu-commands-not-setting-isenabled-properly

Closing this, as there is a "solution" and from my point of view we currently have way more pressing things then to fix issues with the way WPF works.
